modules/v2/main.tf

```diff
@@ -0,0 +1,285 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_cloud_run_v2_service" "main" {
+  provider = google-beta
+
+  project     = var.project_id
+  name        = var.service_name
+  location    = var.location
+  description = var.description
+  labels      = var.service_labels
+
+  template {
+    revision        = var.revision
+    labels          = var.template_labels
+    annotations     = var.template_annotations
+    timeout         = var.timeout
+    service_account = var.service_account
+
+    execution_environment            = var.execution_environment
+    encryption_key                   = var.encryption_key
+    max_instance_request_concurrency = var.max_instance_request_concurrency
+    session_affinity                 = var.session_affinity
+
+    dynamic "scaling" {
```

Comment:
Why are we keeping this as dynamic block? Are we expecting multiple scaling settings! Which one would take precedence?

Reply:
Again, these are not multiple blocks. It's a single object similar to `vpc_access` and others.
There is `scaling` for each revision and overall service. Scaling for overall service splits the instances across revisions.